### ISSUE TITLE
Http request: Control characters are skipped indefinitely (Does not respect limit on max http headers size)

### ISSUE THREAD

**Issue by mhstnsc** [netty/netty#5796]
Steps:
- start an http server
- while true; do printf '\xD'; done | netcat localhost 80

Observed:  The parsing never ends. 

The HTTP standard does not contain any control characters so i wonder what is the reasoning behind this.

https://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html

I traced this to commit fda8808210c4f4f092a93da5929df43062c3aa6c ( Rewrite HttpObjectDecoder to make use of proper state machine)

Maybe @idelpivnitskiy can shed some light. 

The code seems to explain the behavior.

```
    @Override
    protected void decode(ChannelHandlerContext ctx, ByteBuf buffer, List<Object> out) throws Exception {
        if (resetRequested) {
            resetNow();
        }

        switch (currentState) {
        case SKIP_CONTROL_CHARS: {
            if (!skipControlCharacters(buffer)) {
                return;
            }
            currentState = State.READ_INITIAL;
        }
```

**Comment by idelpivnitskiy** [comment]
@mstanescu thanks for the report!
I'll take a look on the weekend

**Comment by normanmaurer** [comment]
@idelpivnitskiy did you have time yet ?

**Comment by creativeprogramming** [comment]
@mstanescu interesting security issue, can you share a PR with your solution to that issue? This may improve approval time as they can test the PR ...

**Comment by normanmaurer** [comment]
@idelpivnitskiy ping...

**Comment by mhstnsc** [comment]
@idelpivnitskiy  @normanmaurer  I looked deeper in the history. Actually the skipping was introduced by @trustin in this commit  9cdc4a959e913b823d59a65368ab1238e5386549

> commit 9cdc4a959e913b823d59a65368ab1238e5386549
> Author: Trustin Lee <trustin@gmail.com>
> Date:   Thu Feb 12 05:02:22 2009 +0000
> 
>     Better white space handling in HTTP

**Comment by Scottmitch** [comment]
this does not sound like intentional behavior and sounds like a vulnerability.

**Comment by andrericos** [comment]
It has already been fixed in #10111 , hasn't it?

**Comment by idelpivnitskiy** [comment]
Yep, #10111 fixed this issue. Now decoder stops with `TooLongFrameException`. Thanks, @andrericos!
